### PR TITLE
Add variant calling training material

### PR DESCRIPTION
**Pull request by lescai**
This PR adds the training material for variant calling applications, using nf-core/sarek as discussed during the nf-core hackathon in Barcelona 2023.
Main features:
- a theory recap on variant calling
- specific config allows nf-core/sarek to run entirely on a gitpod environment
- run through analysis and interpretation of results
As a first draft for material of this type, all suggestions and improvements are welcome.
Note I've added MathJax support as described in mkdocs, in order to enable mathematical formulas in the theory page.

### CODE REVIEW

**Comment by lescai** [comment]
since I have no permissions to request specific reviewers, just tagging here:
- @justinegeffen as discussed
- @vdauwera for a quick review of the GATK theory
- @mribeirodantas for follow up

File: docs/variantcalling/theory.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+### Filtering Variants
+
+There are several ways to spot potential false positives through filtering.
+
+*Hard filtering* uses pre-defined thresholds of different variant annotations (allele-depth, mapping quality and many others) in order to flag variants passing all these criteria, and those failing to meet any of them. This approach is mostly useful when calling a few samples and enough data are not available for more sophisticated solutions.
+
+*Soft filtering* infers the thresholds to be applied from the data themselves. This approach uses the distributions of the annotations, and their overlap with known and validated variants: it defines those combinations of annotations which are more likely to describe true positives (the variants they refer to in the analysis cohort overlap with those validated in other databases). This approach is used by a GATK tool called Variant Quality Score Recalibration (VQSR).
+
+More details can be found on the [GATK VQSR pages](https://gatk.broadinstitute.org/hc/en-us/articles/360035531612-Variant-Quality-Score-Recalibration-VQSR-).
```
Comment (mribeirodantas) [inline docs/variantcalling/theory.md:132]:
```suggestion
More details can be found on the [GATK VQSR page](https://gatk.broadinstitute.org/hc/en-us/articles/360035531612-Variant-Quality-Score-Recalibration-VQSR-).
```

File: docs/variantcalling/sarek.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+One might also need to use custom files: in this case the user might either provide specific parameters at command line, or create a config file adding a new section to the `genome` object. See [here](https://nf-co.re/docs/usage/reference_genomes#custom-genomes) for more details.
+
+We will follow this specific approach in this tutorial, since the data we will be using have been simulated on chromosome 21 of the Human GRCh38 reference, and we have prepared fasta, indexes and annotation files containing only this chromosome locally.
+
+
+### Input files
+
+The input data should be provided in a CSV file, according to a format that is largely common for nf-core pipelines.
+The format is described in the [sarek usage pages](https://nf-co.re/sarek/3.3.2/docs/usage#input-sample-sheet-configurations).
```
Comment (mribeirodantas) [inline docs/variantcalling/sarek.md:40]:
```suggestion
The format is described in the [sarek usage page](https://nf-co.re/sarek/3.3.2/docs/usage#input-sample-sheet-configurations).
```

File: docs/variantcalling/index.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+description: Overview of the Variant Calling Training material
+hide:
+    - toc
+---
+
+# Welcome
+
+Welcome to our Nextflow workshop on variant calling!
+
+In this workshop, we will recap the application of next generation sequencing to identify genetic variations in a genome. You will learn how to use the pipeline [sarek](https://nf-co.re/sarek/3.3.2) to carry out this data-intensive workflow efficiently. We will cover topics such as experimental design, configuration of the pipeline and code execution.
```
[the remark this answers is not in the document]
Comment (lescai) [reply docs/variantcalling/index.md:11]: no that's correct, I guess it's just automatic when ones copy/pastes from the URL 🙈

File: docs/variantcalling/theory.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
+# Calling Variants on Sequencing Data
+
+Before we dive into one of the nf-core pipelines used for variant calling, it's worth looking at some theoretical aspects of variant calling.
+
+## Overview
+
+The term "variant calling" is rooted in the history of DNA sequencing, and it indicates an approach where we identify (i.e. call) positions in a genome (loci) which are variable in a population (genetic variants). The specific genotype of an individual at that variant locus is then assigned.
+
+There are many different approaches for calling variants from sequencing data: here, we will look more specifically at a reference-based variant calling approach, i.e. where a reference genome is needed variant sites and variant sites are identified by comparing the reads to this reference.
```
[the remark this answers is not in the document]
Comment (lescai) [reply docs/variantcalling/theory.md]: corrected

**Comment by mribeirodantas** [comment]
🥳

**Pull request closed without merging**